crawl_OAG.py

```diff
@@ -97,49 +107,188 @@ def crawl_helper(professor, university, file):
-            else:
-                temp_dict["title"] = ""
+        if "title" in pub_json:
+            temp_dict["title"] = pub_json["title"]
+        else:
+            temp_dict["title"] = ""
             
-            if "abstract" in pub_json:
-                temp_dict["abstract"] = pub_json["abstract"]
-            else:
-                temp_dict["abstract"] = ""
+        if "abstract" in pub_json:
+            temp_dict["abstract"] = pub_json["abstract"]
+        else:
+            temp_dict["abstract"] = ""
             
-            if "doi" in pub_json:
-                # print(pub_json["doi"])
-                temp_dict["doi"] = pub_json["doi"]
-            else:
-                temp_dict["doi"] = ""
+        if "doi" in pub_json:
+            temp_dict["doi"] = pub_json["doi"]
+        else:
+            temp_dict["doi"] = ""
             
-            if "citations" in pub_json:
-                temp_dict["citations"] = pub_json["n_citation"]
-            else:
-                temp_dict["citations"] = ""
+        if "citations" in pub_json:
+            temp_dict["citations"] = pub_json["n_citation"]
+        else:
+            temp_dict["citations"] = ""
             
-            publications = publications.append(temp_dict, ignore_index=True)
-            # print(temp_dict)
+        publications = publications.append(temp_dict, ignore_index=True)
     
     return publications
 
+def author_crawler(file):
+    """Crawler helper function that crawls data for authors from a file
+
+    Args:
+        file (str): Path of file
+
+    Returns:
+        authors (pandas dataframe): Data containing the authors' id, name, and organization
+
+    """
+
+    # Initialization
+    column_names = ["id", "name", "org", "pubs"]
+    authors = pd.DataFrame(columns = column_names)
+
+    # Check for empty file path
+    if file == "":
+        return authors
+
+    # Open file
+    file_papers = open(file, 'r')
+
+    # Loop through file.
+    while True:
+        # Get next line from file
+        line = file_papers.readline()
+    
+        # If line is empty end of file is reached
+        if not line:
+            break
+
+        # Load the line into json
+        author_json = json.loads(line)
+        temp_dict = {}   
+
+        if "id" in author_json:
+            temp_dict["id"] = author_json["id"]
+
+        else:
+            temp_dict["id"] = ""
+
+        if "name" in author_json:
+            temp_dict["name"] = author_json["name"]
+
+        else:
+            temp_dict["name"] = ""
+
+        if "orgs" in author_json:
+            temp_dict["org"] = author_json["orgs"]
+
+        else:
+            temp_dict["org"] = ""
+
+        if "pubs" in author_json:
+            temp_dict["pubs"] = author_json["pubs"]
+
+        else:
+            temp_dict["pubs"] = ""
+            
+        authors = authors.append(temp_dict, ignore_index=True)
+    
+    return authors
+
+def open_ssh_tunnel(verbose=False):
```

Comment:
Consider modularizing and extracting the functions `open_ssh_tunnel`, `mysql_connect`, `run_query`, `mysql_disconnect`, and `close_ssh_tunnel` into a separate file as it looks like you reuse them in the `intermediary_database.py` file. 

This way, you don't have to copy past the functions and can simply import them from one file.

Reply:
Added sql_helper.py file to solve this issue.